src/sqlfluff/rules/L052.py

```diff
@@ -193,21 +193,38 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                     anchor_segment = self._handle_trailing_inline_comments(
                         context, anchor_segment
                     )
-
-                    fixes = [
-                        LintFix.replace(
-                            anchor_segment,
-                            [
+                    fixes = []
+                    if anchor_segment == context.segment:
```

Comment:
Nitpick: I think rules are often loose about this, but would it be more precise (and still correct) to instead say:
```suggestion
                    if anchor_segment is context.segment:
```